### PR TITLE
Malyavin hw 15

### PR DESCRIPTION
**Pull request by otecMuxah**

### CODE REVIEW

**Comment by otecMuxah** [comment]
@iliatcymbal i have done all except last (4) task, please review and give your comments

File: src/Numerate/Numerate.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+    if (props.from && props.to) {
+        while (counter < props.to) {
+            array.push(counter++);
+        }
+    }
+    if (props.odd) {
+        array = array.filter((item) => {
+            if (item % 2 !== 0) {
+                return item
```
Comment (iliatcymbal) [inline src/Numerate/Numerate.js]: use shorthand record for arrow function (without return); also filter should return true/false ( return item % 2 !== 0; )

File: src/Numerate/Numerate.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    }
+    if (props.even) {
+        array = array.filter((item) => {
+            if (item % 2 === 0) {
+                return item
+            }
+        })
+    }
+
+    return (<ul>
```
Comment (iliatcymbal) [inline src/Numerate/Numerate.js]: place open and close tags on separate line (separate it from brace):
```
return (
  <div> ....
  </div>
);
```

File: src/Numerate/Numerate.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+        {array.map((item, index) => <li key={index}>{item}</li>)}
+    </ul>)
```
Comment (iliatcymbal) [inline src/Numerate/Numerate.js]: in the end of return statement should be a semicolon

File: src/User/User.js [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+
+export const User = (props) => {
+    console.log(props);
+    return (
+        <div className='user-holder'>
+            <div className="name">{props.name}</div>
+            <div className="surname">{props.surname}</div>
+            <div className="age">{props.age}</div>
+        </div>
+    )
```
Comment (iliatcymbal) [inline src/User/User.js]: semicolon after return

File: src/app.component.js [hunk 5]
```diff
@@ -7,14 +7,9 @@ import { Main } from "./main";
 import { Error } from "./error"
 
 export const App = (
-    <div>
+    <div className={'main-wrapper'}>
```
Comment (iliatcymbal) [inline src/app.component.js]: do not need curly braces

File: src/aside/Aside.js [hunk 6]
```diff
@@ -0,0 +1,11 @@
+import React from 'react';
+
+import { Numerate } from "../Numerate/Numerate";
+
+import './aside.scss';
+
+export const Aside = () => {
+    return (<aside className={'aside'}>
+        <Numerate from='2' to='20' odd/>
+    </aside>)
```
Comment (iliatcymbal) [inline src/aside/Aside.js]: semicolon after return statement;
do not place tags and braces on the same line

File: src/content/Content.js [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+import './content.scss';
+import {User} from "../User/User";
+
+export const Content = () => {
+    return (
+        <div className={"content"}>
+            <Greeting name={'Misha'}/>
+            <User name={'Misha'} surname={'Petrov'} age='28'/>
+        </div>
+    )
```
Comment (iliatcymbal) [inline src/content/Content.js]: semicolon

File: src/error/error.js [hunk 8]
```diff
@@ -3,4 +3,4 @@ import React from 'react';
 export const Error = (props) => {
     console.log(props);
     return (<mark style={ {color: props.color} }>{props.text}</mark>);
```
Comment (iliatcymbal) [inline src/error/error.js]: no need brackets if expression takes one line

File: src/footer/Footer.js [hunk 9]
```diff
@@ -2,9 +2,11 @@ import React from 'react';
...
 import { Navigation } from "../navigation/Navigation";
 
+const navItems = ['Home','Gallery','About','Home'];
+
 export const Footer = () => {
     return (<footer className="footer">
-        <Navigation/>
+        <Navigation nav={ navItems }/>
         <p>footer</p>
     </footer>)
```
Comment (iliatcymbal) [inline src/footer/Footer.js]: round brackets from new line; semicolon after return expression

File: src/greeting/Greeting.js [hunk 10]
```diff
@@ -0,0 +1,26 @@
...
+    }
+    if (18 < hours && hours <= 22) {
+        result = 'Good Evening'
+    }
+    if (props.name.length) {
+        result += `, ${props.name}!` || 'Username'
+    }
+
+    return (<div className={'greeting-block'}>{result}</div>)
+};
```
Comment (iliatcymbal) [inline src/greeting/Greeting.js]: no need brackets; semicolon after return

File: src/header/Header.js [hunk 11]
```diff
@@ -5,4 +5,4 @@ export const Header = () => {
     return (<header className="header">
         <p>header</p>
     </header>)
```
Comment (iliatcymbal) [inline src/header/Header.js]:
```
return (
  <header className="header">
     <p>header</p>
  </header>
);
```

File: src/main/Main.js [hunk 12]
```diff
@@ -1,8 +1,15 @@
...
-    </main>)
-};
\ No newline at end of file
+export class Main extends React.Component {
+    render() {
+        return (<main className="main">
+            <Content />
+            <Aside />
+        </main>)
+    }
```
Comment (iliatcymbal) [inline src/main/Main.js]: do not place tags and brackets on the same line;
in the end of return exression should be a semicolon;

File: src/navigation/Navigation.js [hunk 13]
```diff
@@ -1,13 +1,14 @@
...
     const text = 'Home';
-    const items = ['Home','Gallery','About','Home']
+
     return (
         <nav className="main-nav">
             <ul>
-                {items.map((item,index)=><li key={index}><a href={`/${item.toLowerCase()}`}>{item}</a></li>)}
+                {props.nav.map((item,index)=><li key={index}><a href={`/${item.toLowerCase()}`}>{item}</a></li>)}
             </ul>
         </nav>)
```
Comment (iliatcymbal) [inline src/navigation/Navigation.js]: semicolon

File: src/navigation/Navigation.js [hunk 14]
```diff
@@ -1,13 +1,14 @@
...
-export const Navigation = () => {
+export const Navigation = (props) => {
     const text = 'Home';
-    const items = ['Home','Gallery','About','Home']
+
     return (
         <nav className="main-nav">
             <ul>
-                {items.map((item,index)=><li key={index}><a href={`/${item.toLowerCase()}`}>{item}</a></li>)}
+                {props.nav.map((item,index)=><li key={index}><a href={`/${item.toLowerCase()}`}>{item}</a></li>)}
```
Comment (iliatcymbal) [inline src/navigation/Navigation.js]: process case when nav is undefined

File: src/Numerate/Numerate.js (hunk 1 above)
Comment (otecMuxah) [reply src/Numerate/Numerate.js]: done

File: src/Numerate/Numerate.js (hunk 2 above)
Comment (otecMuxah) [reply src/Numerate/Numerate.js]: done

File: src/Numerate/Numerate.js (hunk 3 above)
Comment (otecMuxah) [reply src/Numerate/Numerate.js]: done

File: src/app.component.js (hunk 5 above)
Comment (otecMuxah) [reply src/app.component.js]: done

File: src/aside/Aside.js (hunk 6 above)
Comment (otecMuxah) [reply src/aside/Aside.js]: done

File: src/navigation/Navigation.js (hunk 13 above)
Comment (otecMuxah) [reply src/navigation/Navigation.js]: done

File: src/navigation/Navigation.js (hunk 14 above)
Comment (otecMuxah) [reply src/navigation/Navigation.js]: done

File: src/main/Main.js (hunk 12 above)
Comment (otecMuxah) [reply src/main/Main.js]: done

File: src/header/Header.js (hunk 11 above)
Comment (otecMuxah) [reply src/header/Header.js]: done

File: src/greeting/Greeting.js (hunk 10 above)
Comment (otecMuxah) [reply src/greeting/Greeting.js]: done

File: src/footer/Footer.js (hunk 9 above)
Comment (otecMuxah) [reply src/footer/Footer.js]: done

File: src/error/error.js (hunk 8 above)
Comment (otecMuxah) [reply src/error/error.js]: done

File: src/content/Content.js (hunk 7 above)
Comment (otecMuxah) [reply src/content/Content.js]: done

**Comment by otecMuxah** [comment]
@iliatcymbal all fixes done and last task done too

**Comment by iliatcymbal** [comment]
Few notes:

- npm script should open browser automatically


- [https://github.com/otecMuxah/Misha-Malyavin-todo/blob/malyavin-hw-15/src/UserList/UserList.js](https://github.com/otecMuxah/Misha-Malyavin-todo/blob/malyavin-hw-15/src/UserList/UserList.js)
```
props.list.map(item => (
    <li>
      <User name={item.name} surname={item.surname} age={item.age} />
    </li>
  ));
```
Each child in an array or iterator should have a unique "key" prop.


- [https://github.com/otecMuxah/Misha-Malyavin-todo/blob/malyavin-hw-15/src/navigation/Navigation.js](https://github.com/otecMuxah/Misha-Malyavin-todo/blob/malyavin-hw-15/src/navigation/Navigation.js)
```
props.nav ?
          props.nav.map
```
It is better to use `&&` operator instead of `? : `


- Each attribute form new line
```
<Numerate from="2" to="20" odd />
```


Fix that and merge into master

**Pull request merged**